src/components/Accordion.js

```diff
@@ -1,75 +1,98 @@
+    return dom.div(
+      { className: "accordion" },
+      this.props.children
+    );
+  }
+});
+
+const AccordionPane = React.createClass({
+  propTypes: {
+    item: PropTypes.object,
+    opened: PropTypes.bool,
+    children: PropTypes.array,
+    buttons: PropTypes.object,
+    header: PropTypes.string
   },
 
-  handleHeaderClick: function(i) {
-    const opened = [...this.state.opened];
-    const created = [...this.state.created];
-    const item = this.props.items[i];
+  displayName: "AccordionPane",
+
+  getInitialState: function() {
+    return {
+      opened: this.props.opened
+    };
+  },
 
-    opened[i] = !opened[i];
-    created[i] = true;
+  handleHeaderClick: function() {
+    const opened = this.state.opened;
+    opened = !opened;
+    this.setState({ opened });
+  },
 
-    if (opened[i] && item.onOpened) {
-      item.onOpened();
+  renderButtons(buttons) {
+    if (!buttons) {
+      return null;
     }
 
-    this.setState({ opened, created });
+    return dom.span(
+      { className: "header-buttons" },
+      buttons
+    );
   },
 
-  renderContainer: function(item, i) {
-    const { opened, created } = this.state;
-    const containerClassName =
-          item.header.toLowerCase().replace(/\s/g, "-") + "-pane";
-
-    return div(
-      { className: containerClassName, key: i },
+  renderContent() {
+    const opened = this.state.opened;
+    const children = this.props.children;
 
-      div(
-        { className: "_header",
-          onClick: () => this.handleHeaderClick(i) },
-        Svg("arrow", { className: opened[i] ? "expanded" : "" }),
-        item.header,
-        item.buttons ?
-        dom.span({ className: "header-buttons" },
-          item.buttons.map((button, id) => span({ key: id }, button))
-        ) :
-        null
-      ),
+    if (!opened) {
+      return null;
+    }
 
-      (created[i] || opened[i]) ?
-        div(
-          { className: "_content",
-            style: { display: opened[i] ? "block" : "none" }
-          },
-          React.createElement(item.component, item.componentProps || {})
-        ) :
-        null
+    return dom.div(
+      { className: "_content",
+        style: { display: opened ? "block" : "none" }
+      },
+      children
     );
   },
 
   render: function() {
-    return div(
-      { className: "accordion" },
-      this.props.items.map(this.renderContainer)
+    const { buttons, header } = this.props;
+    const { opened } = this.state;
+
+    const containerClassName =
+          header.toLowerCase().replace(/\s/g, "-") + "-pane";
+
+    return dom.div(
+      { className: containerClassName },
+
+      dom.div(
+        { className: "_header",
+          onClick: this.handleHeaderClick
+        },
+        Svg("arrow", { className: opened ? "expanded" : "" }),
+        header,
+        this.renderButtons(buttons)
+      ),
+      this.renderContent()
     );
   }
 });
 
-module.exports = Accordion;
+module.exports = {
+  AccordionPane,
```

Comment:
Don't export 2 components from the same file. A lot of React tooling expects a single component per file and it's a good pattern. What is the advantage of `AccordionPane` anyway? We can achieve the same thing without an additional component, and this API eagerly renders all of the panes which we don't want.

Perhaps the only awkward thing about passing in an array of items is that we don't support `null` in the array, so you could conditionally include things like `Expressions`. I suggest simply allowing `null`s in the array but keeping the previous API.

Otherwise this PR looks fine. I don't know much about `Expressions` so don't have much to comment there.